src/spec_swagger/schema.cljc

```diff
@@ -0,0 +1,43 @@
+(ns spec-swagger.schema
+  "Generate a Swagger schema from a clojure.spec spec.
+
+  If you want to generate a JSON Schema, look at spec-tools.json-schema. Swagger
+  schemas are based on JSON Schema, but it lacks some of JSON Schema's features
+  while having some that are not found in JSON Schema."
+  (:require [spec-tools.json-schema :as json-schema]
+            [spec-tools.visitor :as visitor :refer [visit]]))
+
+(defn- unwrap
+  "Unwrap [x] to x. Asserts that coll has exactly one element."
+  [coll]
+  {:pre [(= 1 (count coll))]}
+  (first coll))
+
+(defn- spec-dispatch [dispatch spec children] dispatch)
+(defmulti accept-spec spec-dispatch :default ::default)
+
+(defmethod accept-spec 'clojure.core/float? [_ _ _] {:type "number" :format "float"})
+(defmethod accept-spec 'clojure.core/double? [_ _ _] {:type "number" :format "double"})
+
+(defmethod accept-spec 'clojure.spec/or [dispatch spec children]
+  ;; :anyOf is not supported by Swagger 2.0, so we just give up. In principle,
+  ;; we could do better in some special cases. For example, a reasonable schema
+  ;; for (s/or ::int int? ::str string?) would be {:type ["number", "string"]}.
+  {})
```

Comment:
Kinda, but as a convention it's ok. Spec uses internally similar logic - generator is taken from the first value of an `s/or`. Same for conforming.